resourcemanager/features/user_specified_segments.go

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package features
+
+// TreatUserSpecifiedSegmentsAsCaseInsensitive is a feature-toggle which specifies whether User Specified
+// Resource ID Segments should be compared case-insensitively as required.
+//
+// @tombuildsstuff: whilst this IS EXPOSED in the public interface - this is NOT READY FOR USE and should
+// not be exposed in user-focused logic at this time, else this'll become a source of knock-on problems
```

Comment:
Could you explain what you mean by user-focused logic here?

Reply:
Here I'm using the phrase `user-focused logic` to mean "surfaced to end-users" - for example exposing a feature-toggle to do so in the Provider, or setting this to true/doing this by default.

Perhaps this'd be better rephrased as:

```suggestion
// not be exposed to users (i.e. as a feature-toggle) until this work is completed - as this'll become a source of knock-on problems
```

WDYT?

---

An aside: initially I was trying to make the public surface for this a function (e.g. `features.TreatUserSpecifiedSegmentsAsCaseInsensitive()`, which in-turn set a feature-flag within an `internal` package in this project) - to ensure that we don't unintentionally use this toggle outside of this project - but I don't _think_ having this publicly accessible is the end of the world at this point-in-time.

Reply:
I agree, the rephrased comment makes the intention clearer 👍

Reply:
👍 committed this suggestion then